src/tapis-app/Streams/MeasurementsListing/_Layout/Layout.tsx

```diff
@@ -0,0 +1,63 @@
+import { useList } from 'tapis-hooks/streams/measurements';
+import Measurements from '../_components/Measurements';
+import React, { useCallback, useState } from 'react';
+import styles from './Layout.module.scss';
+import { QueryWrapper } from 'tapis-ui/_wrappers';
+
+const Layout: React.FC<{
+  projectId: string;
+  siteId: string;
+  instrumentId: string;
+}> = ({ projectId, siteId, instrumentId }) => {
+  const { data, isLoading, error } = useList({
+    projectId,
+    siteId,
+    instId: instrumentId,
+  });
+
+  const { instrument, site, measurements_in_file, ...measurements } =
+    data?.result ?? {};
+
+  const variables = Object.keys(measurements);
+  const [selected, setSelected] = useState<number>(-1);
+
+  let select = useCallback(
+    (index: number) => {
+      return () => {
+        setSelected(selected === index ? -1 : index);
+      };
+    },
+    //only need to update select function if measurements changed (only time indexes will change)
+    /* eslint-disable-next-line */
+    [measurements]
```

Comment:
Should setSelected be a dependency? Is there a negative impact on rendering if it is?

Reply:
There should be no reason to regenerate this function if setSelected changes. The function returns a function that calls setSelected. The code that uses setSelected is not rerun on a rerender. When this function is executed (whenever an element is clicked) the function referenced by the setSelected variable will be used. Updating the outer function when this occurs will have no effect and may lead to unnecessary rerenders.